translator/source_code/icone_translator.py

```diff
@@ -136,29 +190,105 @@ def get_event_status(start_time_string,end_time_string):
     return event_status
 
 
-# Parse Icone Incident to WZDx
-def parse_incident(incident):
-
-
-
+def create_description(incident):
+    description = incident['description']
+
+    if incident.get('sensor'):
+        description += '\n sensors: '
+        for sensor in incident['sensor']:
+            if not isinstance(sensor, str):
+                if sensor['@type'] == 'iCone':
+                    description += '\n' + json.dumps(parse_icone_sensor(sensor), indent=2)
+            else:
+                sensor = incident['sensor']
+                if sensor['@type'] == 'iCone':
+                    description += '\n' + json.dumps(parse_icone_sensor(sensor), indent=2)
```

Comment:
What case is this handling? The sensor appears to always be an object and not a string so the check on 199 should always go through right? Am I missing a case?

Reply:
Short answer: this is for a very specific but required case

Long answer: We are using the xmltodict library to read xml into a dictionary object in python. In xml, you can have multiple tags with the same name, and this occurs a lot in the incidents_extended.xml iCone data. Dictionaries must have unique keys, so when this happens the xmltodict library imports these objects as a list of objects. We decided to use a for loop to iterate through this list of objects. Inconveniently, when there is only a single key in the xml, the xmltodict library takes it in as a dictionary and not a list, and then the for loop iterates over the dictionary, reading invididual keys and values, not full dictionary objects like we were expecting. To combat this, we check if the iterated object is a string. If it is, then we access it's parent as the object and do the same thing we would have done.

Hopefully that made sense

Reply:
Well that makes more sense now. Not the most helpful thing for xmltodict to behave differently with the data like that, but at least I see why the extra case handling here.